File: ios/Yomiyasu/Core/Models/UserWordModels.swift
import Foundation

struct UserWord: Decodable, Sendable, Identifiable, Equatable {
    let word: String
    let display: String
    let sentence: String
    let meaning: [String]
    let reading: String
    let frequency: Int
    let pitch: [Int]
    let createdAt: Date?

    var id: String { word }
}

enum WordsSort: String, CaseIterable, Identifiable {
    case newest
    case oldest = "new"
    case frequencyDesc = "frequency"
    case frequencyAsc = "!frequency"

    var id: String { rawValue }

    var title: String {
        switch self {
        case .newest: "Recientes"
        case .oldest: "Antiguas"
        case .frequencyDesc: "Más frecuentes"
        case .frequencyAsc: "Menos frecuentes"
        }
    }

    var queryValue: String? {
        switch self {
        case .newest: nil
        case .oldest, .frequencyDesc, .frequencyAsc: rawValue
        }
    }
}

extension UserWord {
    var frequencyLabel: String {
        switch frequency {
        case 0: "—"
        case ..<1500: "Muy alta"
        case ..<5000: "Alta"
        case ..<15000: "Media"
        case ..<30000: "Baja"
        default: "Muy baja"
        }
    }
}

File: android/app/src/main/java/es/manabe/yomiyasu/core/models/UserWordModels.kt
package es.manabe.yomiyasu.core.models

import kotlinx.serialization.Serializable

@Serializable
data class UserWord(
    val word: String,
    val display: String = "",
    val sentence: String = "",
    val meaning: List<String> = emptyList(),
    val reading: String = "",
    val frequency: Int = 0,
    val pitch: List<Int> = emptyList(),
    val createdAt: String? = null,
) {
    val frequencyLabel: String
        get() = when {
            frequency == 0 -> "—"
            frequency < 1500 -> "Muy alta"
            frequency < 5000 -> "Alta"
            frequency < 15000 -> "Media"
            frequency < 30000 -> "Baja"
            else -> "Muy baja"
        }
}

enum class WordsSort(val rawValue: String?) {
    Newest(null),
    Oldest("new"),
    FrequencyDesc("frequency"),
    FrequencyAsc("!frequency"),
    ;

    val title: String
        get() = when (this) {
            Newest -> "Recientes"
            Oldest -> "Antiguas"
            FrequencyDesc -> "Más frecuentes"
            FrequencyAsc -> "Menos frecuentes"
        }

    val queryValue: String?
        get() = rawValue
}
